ai_edge_torch/generative/layers/model_config.py

```diff
@@ -55,9 +55,10 @@ class FeedForwardType(enum.Enum):
 
 @dataclass
 class AttentionConfig:
-  """Attention moduel's parameters."""
+  """Attention model's parameters."""
 
   num_heads: int
+  head_dim: int
```

Comment:
i guess in most common cases (other than OpenELM) the `head_dim` could be automatically derived as the current library does, this will eliminate setting one more parameter when constructing the model.

Can we keep the current behavior (so that most models don't need a change and supply this additional parameter), but allow the parameter to be overwritten, or derived in another user supplied lamda?

Reply:
this is kinda important design decision if we want to keep the `ModelConfig`api easy and simple to use, without exposing too many knobs to the user.

Reply:
@talumbau @yichunk @hheydary any thoughts you could share here?

Reply:
GemmaV2 needs this exact change and I was about to make the changes to CausalSelfAttention that Maxime does here. I suspect more architectures will use this trick to squeeze down the parameter count a bit. So, I am +1 to being explicit here.

Reply:
cool, sgtm. If this is going to be common, exposing it explicitly sounds like a good plan.